### ISSUE TITLE
Bug Build pf-host-agent-8.12.2 | pf-elastic-symbolizer | pf-elastic-collector

### ISSUE THREAD

**Issue by yakhatape** [elastic/elastic-agent#4455]
Hi ,

I identify an issue into some build package cf. Title of the issue. Which are build with empty string falg which broken some mirroring package integration.

As its recommended by the Linux Foundation and from the native rpm build tools, flag should be not set if empty (to be reported as NONE value) instead of empty string (reported as NULL Value): 
https://refspecs.linuxfoundation.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic/swinstall.html
https://rpm-software-management.github.io/rpm/manual/tags.html

I think the problem come from the nfpm tools when the flag isnt set properly on the build side/script.

We can see it directly on package rpm level : 

```
rpm --nosignature -qp redhat/1/d82/pf-elastic-symbolizer/0:8.12.2-1/x86_64/d8287a4a64d7685f4cd712a91175d19ea10281f29701f728098472af819c9f8f/pf-elastic-symbolizer-8.12.2-1.x86_64.rpm --qf "$(rpm --querytags | sed -nr 's/(.*)/\1:%{\1}/p' | tr '[:space:]' '\n')"
rpm --nosignature -qp redhat/1/55d/pf-host-agent/0:8.12.2-1/x86_64/55d81eebcee75e947d56c612a6fbe5feef7a57e63a1500fa1a1f0b71ae89fb7d/pf-host-agent-8.12.2-1.x86_64.rpm --qf "$(rpm --querytags | sed -nr 's/(.*)/\1:%{\1}/p' | tr '[:space:]' '\n')"
rpm --nosignature -qp redhat/1/3f7/pf-elastic-collector/0:8.12.2-1/x86_64/3f7d16d24caae35138ccc6b63168a3756a66851dd1589877bfff6bd854003d6d/pf-elastic-collector-8.12.2-1.x86_64.rpm --qf "$(rpm --querytags | sed -nr 's/(.*)/\1:%{\1}/p' | tr '[:space:]' '\n')"
```

All of them return the following : 
![image](https://github.com/elastic/elastic-agent/assets/9256016/4d7b60b0-c14f-4f6c-9740-19c95aee7c21)

```
rpm --nosignature -qp6fb/elasticsearch/0:8.12.2-1/x86_64/6fbd2bc816b367e306a880e384349aec5fdcfe7b4d43ad6301027d3d911a646a/elasticsearch-8.12.2-1.x86_64.rpm --qf "$(rpm --querytags | sed -nr 's/(.*)/\1:%{\1}/p' | tr '[:space:]' '\n')"
```

If I check in parallel the elasticsearch package for the same version I can see the Flag correctly setup : 
![image](https://github.com/elastic/elastic-agent/assets/9256016/276f6d13-c443-4746-b956-cc68c065e385)

This behavior about the Vendor empty string Flag impact the import process into database which check if a tag is NONE of not. When the flag is not NONE and its an empty string its provoc an SQL query error of type VIOLATION CONSTRAINT : 
a NULL value violates the NOT NULL constraint of the column

Can you please fix this issue ? And is it possible to rebuild all previous package impacted by this empty string Flag on rpm packages ?

Thanks in advance

**Comment by christos68k** [comment]
We fixed this issue internally and the fix is part of 8.13.0 and also backported to 8.12 which will make it part of 8.12.3. However, we can't rebuild older packages to include it.

**Comment by yakhatape** [comment]
Happy to help
